modules/seqexec/server/src/main/scala/seqexec/server/nifs/Nifs.scala

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.nifs
+
+import cats.data.Reader
+import cats.effect.Sync
+import cats.effect.IO
+import cats.effect.LiftIO
+import cats.implicits._
+import edu.gemini.spModel.config2.Config
+import edu.gemini.spModel.gemini.nifs.InstNIFS._
+import edu.gemini.spModel.gemini.nifs.InstEngNifs._
+import gem.enum.LightSinkName
+import java.lang.{ Double => JDouble }
+import java.lang.{ Integer => JInt }
+import java.beans.PropertyDescriptor
+import shapeless.tag
+import shapeless.tag.@@
+import seqexec.server.ConfigUtilOps._
+import seqexec.model.dhs.ImageFileId
+import seqexec.model.enum.Instrument
+import seqexec.model.enum.Resource
+import seqexec.server.ConfigUtilOps.ExtractFailure
+import seqexec.server.ConfigResult
+import seqexec.server.InstrumentSystem
+import seqexec.server.ObserveCommand
+import seqexec.server.Progress
+import seqexec.server.SeqActionF
+import seqexec.server.SeqObserveF
+import seqexec.server.TrySeq
+import seqexec.server.keywords.DhsClient
+import seqexec.server.keywords.DhsInstrument
+import seqexec.server.keywords.KeywordsClient
+import seqexec.server.InstrumentSystem.AbortObserveCmd
+import seqexec.server.InstrumentSystem.InfraredControl
+import seqexec.server.InstrumentSystem.StopObserveCmd
+import seqexec.server.nifs.NifsController._
+import squants.Time
+import squants.time.TimeConversions._
+
+final case class Nifs[F[_]: LiftIO: Sync](controller: NifsController[IO], // TODO Convert to NifsController[F]
```

Comment:
I'd like to convert this to tagless style but this particular change cascaded to a very large change